src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -277,7 +277,7 @@ public ProjectContext Build()
 
                     if (platformDependency != null)
                     {
-                        platformLibrary = libraries[new LibraryKey(platformDependency.Value.Name)];
+                        libraries.TryGetValue(new LibraryKey(platformDependency.Value.Name), out platformLibrary);
```

Comment:
When this happens, isn't this an error state? If this comes back null, should we throw here, print a warning, catch it in the command resolver and then return null from that resolver?